src/main/java/org/json/XMLParserConfiguration.java

```diff
@@ -321,8 +326,8 @@ public int getMaxNestingDepth() {
 
     /**
      * Defines the maximum nesting depth that the parser will descend before throwing an exception
-     * when parsing the XML into JSON. The default max nesting depth is undefined, which means the
-     * parser will go as deep as the maximum call stack size allows. Using any negative value as a
+     * when parsing the XML into JSON. The default max nesting depth is 512, which means the parser
+     * will go as deep as the maximum call stack size allows. Using any negative value as a
```

Comment:
This comment is slightly incorrect - now we have a default limit of 512, so the parser **won't** `go as deep as the maximum call stack size allows`, instead it will stop at the depth limit and throw a JsonException.  
You can rephrase it like this:
```
The default max nesting depth is 512, which means the parser will throw a JsonException 
if the maximum depth is reached.  
Using any negative value as a parameter is equivalent to setting no limit to the nesting depth, 
which means the parses will go as deep as the maximum call stack size allows. 
```